design/specs/table_inspector.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Table Inspector
+description: 
+published: true
+date: 2022-08-02
+tags: 
+editor: markdown
+dateCreated: 2022-08-02
+---
+
+## Context
+
+Using the table inspector, users can access information and configuration options for each individual table component, including columns, rows, and cells.
+
+### Design Goals
+
+- Provide easy access to table component configurations.
+- Make features easier to discover.
+
+## Enabling the Table Inspector
+
+The table inspector can be enabled or disabled manually or automatically based on triggers.
+
+To enable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to show the table inspector.
+- Open a newly created table, as part of another flow like [data import](#data-import) or [column extraction](#column-extraction).
+
+When users enable the table inspector, a panel will appear on the right side of the screen, sharing the same container as the table and pushing its contents rather than being stacked on top of it.
+
+- In its default mode, the inspector will provide information and configurations for the current table.
+- The inspector provides additional information and configuration options depending on what's currently selected.
+
+To disable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to hide the table inspector.
+
+## Using the Table Inspector
+
+Whatever is presently selected will determine the content of the table inspector. During non-selection mode, the inspector simply displays properties for the current table.
+
+### Table Inspector Modes
+
+The current selection is what triggers the various inspection modes. The inspector panel's header contains a brief description of what is included in the selection.
+
+### Table Properties
+
+Shown during non-selection mode.
+
+Table properties include:
+
+- Table Name and Description
+- Table Links
+- Table Display Options
+  - Record Summary
+  
+![image](/assets/design/specs/table_inspector/182566720-3622f073-24ee-4aff-8460-64442f271fea.png)
+
+### Column Properties
+
+Shown when a [column](#column-selection) or [cell](#cell-selection) are selected.
+
+Column properties include:
+
+- Column Name
+- Column Constraints
+  - Allow NULL
+  - Allow Duplicates
+- Column Data Type
+  - Select Type
+  - Database Options
+  - Format Options
+  
+![image](/assets/design/specs/table_inspector/182350473-c20ebabb-2a41-4bf7-ad50-dfda965b1081.png)
+
+### Link Column Properties
+
+Shown when a foreign key column or cell is selected.
+
+Link column properties include:
+
+- Column Name
+- Link Information
+  - Table source
+  - Column source
+  
+![image](/assets/design/specs/table_inspector/182350542-a5af060c-16f7-40b5-9a00-b7b6cc32e2a2.png)
+
+### Multiple Columns
+
+Shown when multiple columns are selected.
+
+Multiple column properties:
+
+- Data Type
```

Comment:
This option won't always be possible right since the multiple columns selected can be of two different types.